ui/server/src/main/scala/pl/touk/nussknacker/ui/api/PropertiesResources.scala

```diff
@@ -0,0 +1,116 @@
+package pl.touk.nussknacker.ui.api
+
+import akka.http.scaladsl.server.Route
+import cats.data.OptionT
+import cats.instances.future._
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.JsonCodec
+import io.circe.generic.extras.semiauto.deriveConfiguredDecoder
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.additionalInfo.{AdditionalInfo, AdditionalInfoProvider}
+import pl.touk.nussknacker.engine.api.{CirceUtil, MetaData}
+import pl.touk.nussknacker.engine.api.context.{ProcessCompilationError, ValidationContext}
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.util.loader.ScalaServiceLoader
+import pl.touk.nussknacker.restmodel.validation.ValidationResults.NodeValidationError
+import pl.touk.nussknacker.ui.process.processingtypedata.ProcessingTypeDataProvider
+import pl.touk.nussknacker.ui.process.repository.FetchingProcessRepository
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+import pl.touk.nussknacker.restmodel.displayedgraph.{DisplayableProcess, ProcessProperties}
+import org.springframework.util.ClassUtils
+import pl.touk.nussknacker.engine.api.CirceUtil._
+import pl.touk.nussknacker.engine.api.typed.TypingResultDecoder
+import pl.touk.nussknacker.engine.variables.GlobalVariablesPreparer
+import pl.touk.nussknacker.restmodel.definition.UIParameter
+import pl.touk.nussknacker.engine.api.component.AdditionalPropertyConfig
+import pl.touk.nussknacker.restmodel.process.ProcessingType
+import pl.touk.nussknacker.ui.process.subprocess.SubprocessRepository
+import pl.touk.nussknacker.ui.validation.AdditionalPropertiesValidator
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * This class should contain operations invoked for each node (e.g. node validation, retrieving additional data etc.)
+  */
+class PropertiesResources(val processRepository: FetchingProcessRepository[Future],
```

Comment:
I wonder if it's necessary to create new resource for that? Maybe we can merge it with NodeResources, they do the same operations, it'll be easier to keep them in sync?